MSAL/src/requests/MSALInteractiveRequest.m

```diff
@@ -151,7 +151,12 @@ - (void)acquireTokenImpl:(MSALCompletionBlock)completionBlock
             {
                 _code = oauthResponse.authorizationCode;
 
-                // TODO: handle MSIDWebOAuth2Response and instance aware flow (cloud host)
+                // handle instance aware flow (cloud host)
+                if ([response isKindOfClass:MSIDWebAADAuthResponse.class])
+                {
+                    MSIDWebAADAuthResponse *aadResponse = (MSIDWebAADAuthResponse *)response;
+                    [_parameters setCloudAuthorityWithCloudHostName:aadResponse.cloudHostName];
```

Comment:
do we need to save the whole cloudAuthority in parameters if we only use host from it?

Reply:
In `[MSALRequestParameters msidConfiguration]` it needs the whole `cloudAuthority` :)